src/vm/amd64/umthunkstub.S

```diff
@@ -153,6 +157,21 @@ LOCAL_LABEL(DoThreadSetup):
         call            C_FUNC(CreateThreadBlockThrow)
         jmp             LOCAL_LABEL(HaveThread)
         
+LOCAL_LABEL(InvalidTransition):
```

Comment:
This is not correct for Unix. There is no homing of argument registers and the argument registers are different from Windows. See the stack layout starting at line 44 in this file.